src/lib/server/orders.ts

```diff
@@ -1514,13 +1514,54 @@ export async function updateAfterOrderPaid(order: Order, session: ClientSession)
 						}))
 				  );
 
-		await collections.challenges.updateOne(
-			{ _id: challenge._id },
-			{
-				$inc: { progress: increase }
-			},
-			{ session }
-		);
+		if (increase) {
+			await collections.challenges.updateMany(
```

Comment:
Ah sorry it was `updateOne`, it was correct before

because it's in a loop (`for const challenge of challenges`)

Still, then the question why @Tirodem only had one challenge updatd

Reply:
About this one, it was an old challenge with no product whitelist.
Now I've deleted it and no challenge has incrementation on last tries.